src/game.zig

```diff
@@ -689,6 +693,93 @@ pub const World = struct { // MARK: World
 		main.audio.setMusic(self.playerBiome.raw.preferredMusic);
 	}
 
+	fn findValidFolderName(allocator: main.heap.NeverFailingAllocator, name: []const u8) []const u8 {
+		// Remove illegal ASCII characters:
+		const escapedName = main.stackAllocator.alloc(u8, name.len);
+		defer main.stackAllocator.free(escapedName);
+		for(name, 0..) |char, i| {
+			escapedName[i] = switch(char) {
+				'a'...'z', 'A'...'Z', '0'...'9', '_', '-', '.', ' ' => char,
+				128...255 => char,
+				else => '-',
+			};
+		}
+
+		// Avoid duplicates:
+		var resultName = main.stackAllocator.dupe(u8, escapedName);
+		defer main.stackAllocator.free(resultName);
+		var i: usize = 0;
+		while(true) {
+			const resultPath = std.fmt.allocPrint(main.stackAllocator.allocator, "saves/{s}", .{resultName}) catch unreachable;
+			defer main.stackAllocator.free(resultPath);
+
+			if(!main.files.cubyzDir().hasDir(resultPath)) break;
+
+			main.stackAllocator.free(resultName);
+			resultName = std.fmt.allocPrint(main.stackAllocator.allocator, "{s}_{}", .{escapedName, i}) catch unreachable;
+			i += 1;
+		}
+		return allocator.dupe(u8, resultName);
+	}
+	pub fn tryCreateWorld(worldName: []const u8, worldSettings: WorldSettings) !void {
```

Comment:
I think logically this would better fit in `src/server/world.zig` since it is creating the (server) world.
This `World` here is only used on the client.